### PR TITLE
Update PutMetric API change

### PR DESCRIPTION
**Pull request by trungnq97**

### CODE REVIEW

File: src/test/java/com/beeinstant/metrics/MetricsManagerTest.java [hunk 1]
```diff
@@ -62,39 +64,42 @@ public void tearDown() throws IOException {
...
-        MetricsManager.flushAll();
+        MetricsManager.getRootMetricsLogger().flush(9999);
+        MetricsManager.flushAll(9999);
         responseProvider.verify();
     }
 
     @Test
     public void testExtendInvalidDimensionsIgnoreAndReportError() throws UnsatisfiedExpectationException {
-        responseProvider.expect(POST, "/PutMetric", "application/json", "{\"metrics\":\"d.service=ImageSharing,m.MetricErrors=2\n\"}").respondWith(200, "application/json", "");
+        responseProvider.expect(POST, "/PutMetric?signature=8XWXymIrwhCNni6p7U9jSskFXe95qNvtAvmb5%2FE7eA4%3D&publicKey=PublicKey&timestamp=9999", "text/plain", "d.service=ImageSharing,m.MetricErrors=2\n").respondWith(200, "application/json", "");
```
Comment (linhdang) [inline src/test/java/com/beeinstant/metrics/MetricsManagerTest.java:98]: security issue?

Comment (trungnq97) [reply src/test/java/com/beeinstant/metrics/MetricsManagerTest.java:98]: What's the security issue?

**Comment by linhdang** [comment]
all the encoding key can be seen from access log or from the sniffing tool

**Comment by trungnq97** [comment]
1) Connection from SDK to BeeInstant backend is HTTPS.
2) The only key in the url is publicKey (supposed to be public).
3) Signature is for authentication & integrity check. It's generated by using secretKey which is not transmitted.

**Pull request merged**